### ISSUE TITLE
nextCloud 15?

### ISSUE THREAD

**Issue by BloodyIron** [Yetangitu/owncloud-apps#114]
Any chance we can get this compatible with nextCloud 15? Please? Without having to fork this? :/

**Comment by 4oo4** [comment]
@BloodyIron All you should have to do is to edit `appinfo/info.xml` of both apps to have the max-version for Nextcloud set to 15.0, and they both seem to work:

```
    <dependencies>
        <nextcloud min-version="8.1" max-version="15.0"/>
                <database>pgsql</database>
                <database>sqlite</database>
                <database>mysql</database>
    </dependencies>
```

**Comment by BloodyIron** [comment]
If it's such little work, why can't the releaser dev address it then?

**Comment by rlitman** [comment]
Because when I try that on my install, it causes Nextcloud to die with an Internal Server Error.
I tried the patch from [here](https://github.com/Yetangitu/owncloud-apps/pull/108):
but Nextcloud complained automatically disabled the app, because it "made Nextcloud unstable"

**Comment by BloodyIron** [comment]
Sounds like it's not that little work then...

**Comment by reiner-text** [comment]
> Part of having a Nextcloud is that you manage it yourself. In the time it took you to make this issue, you could have filed a PR and/or already fixed it for yourself. If you don't like the developer's response time, then fork it.

@4oo4 Not every (Nextcloud|GitHub) user is such a brilliant developer like you and can fix it for himself.
There's no simple change for ``files_reader`` from ``13`` to ``15`` in ``info.xml`` because the error message ``Error: This app can not be enabled because it makes the server unstable`` pops up.

Not every Nextcloud admin knows which fork of this app works for now. Apart from the signing certificate.

After more than one year without any change and support it seems to me that this app is no longer maintained. The author should drop a line in the ``README.md`` file and the repo header.

**Comment by BloodyIron** [comment]
:/ is there any other way to read epub or other such formats in nextCloud then?

**Comment by reiner-text** [comment]
IMHO with OnlyOffice only, but for that you have to install a Docker container and so this solution for Shared Hosting is as good as not usable.

**Comment by 4oo4** [comment]
@reiner-text I agree that the `README.md` needs to be updated to reflect that it's unmaintained, but I deleted the comment you're quoting because it was rude and unhelpful. A part of self-hosting does involve personal responsibility (and encourages tinkering like that), but you're right that my hack definitely isn't a replacement for the maintenance that this needs, it was just something that happened to work for me.

I'm not a developer, just a self-hosting nerd like everyone else, and I'm sorry for bringing that garbage into this thread.

**Comment by BloodyIron** [comment]
I hope we can find some way to bring this back, because I think nextCloud as a personal library, for epub and so much more, is just a really awesome idea. And I'm really bummed out about it not working :(

**Comment by BloodyIron** [comment]
Bump? Anyone prepared to fork?

**Comment by BloodyIron** [comment]
Tried the manual override to xml for v15 and it made my nextCloud completely unstable, had to manually disable the app with OCC CLI.

 :(

**Comment by reiner-text** [comment]
It's an abandoned app, try [Ebook reader](https://apps.nextcloud.com/apps/files_ebookreader) instead. 😉

**Comment by rlitman** [comment]
Thanks, just did.  Ebook reader is  a little clunky, but for a 0.0.1 release, I would say that it has exceeded my expectations, and works well enough.

**Comment by reiner-text** [comment]
It's a good replacement for ebook displaying.

> When this application is enabled publicly shared ebook documents will also get shown in the ebook reader instead of only showing a single static snapshot of the document.

This doesn't work for me, unfortunately.

**Comment by rlitman** [comment]
I tried it with a few epub files.  There is no way to flip pages with the mouse, but you can get to the table of contents by clicking on the hamburger, and you can use your cursor keys to flip pages.  I am not sure how that would work in a mobile interface.
